### PR TITLE
Lecture tdo

### PR DESCRIPTION
**Pull request by domino468**

### CODE REVIEW

File: src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java [hunk 1]
```diff
@@ -17,31 +18,48 @@
     @Value("${description.length.limit}")
     private int descriptionLengthLimit;
 
-    LectureService(LectureRepository lectureRepository, CourseRepository courseRepository) {
-        this.lectureRepository = lectureRepository;
-        this.courseRepository = courseRepository;
+    LectureService(LectureRepository lectureRepository, CourseRepository courseRepository) { this.lectureRepository = lectureRepository;this.courseRepository = courseRepository; }
+
+    LectureDTO convertLectureTOTDO(Lecture lecture){
```
Comment (adamkulawik) [inline src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java:23]: `convertLectureToDTO`

File: src/main/java/com/coursesPlatform/coursePortfolio/LectureDTO.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
+package com.coursesPlatform.coursePortfolio;
+
+import java.math.BigDecimal;
+import java.util.Objects;
+
+public class LectureDTO {
```
Comment (adamkulawik) [inline src/main/java/com/coursesPlatform/coursePortfolio/LectureDTO.java:6]: Check if this is can be package private pls

File: src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java [hunk 3]
```diff
@@ -17,31 +18,48 @@
...
+        return  new LectureDTO(
+                lecture.getTitle(),
+                lecture.getDescription(),
+                lecture.getPrice(),
+                lecture.getDuration()
+        );
+    }
```
Comment (adamkulawik) [inline src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java:30]: Missing empty line

File: src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java [hunk 4]
```diff
@@ -17,31 +18,48 @@
...
+    Lecture convertLectureDTOToLecture(LectureDTO lectureDTO){
+        Lecture lecture = new Lecture();
+        lecture.setTitle(lectureDTO.getTitle());
+        lecture.setDescription(lectureDTO.getDescription());
+        lecture.setDuration(lectureDTO.getDuration());
+        lecture.setPrice(lectureDTO.getPrice());
+        return lecture;
     }
+    List<LectureDTO> findAllLectures(){
```
Comment (adamkulawik) [inline src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java:39]: same here

File: src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java [hunk 5]
```diff
@@ -17,31 +18,48 @@
...
-        if ( lecture.getDescription().length() >= descriptionLengthLimit ) {
+        if ( lectureDTO.getDescription().length() >= descriptionLengthLimit ) {
             throw new IllegalLengthException();
         }
-
-        course.getLectures().add(lecture);
-        return lectureRepository.save(lecture);
+        Lecture lecture1 = lectureRepository.save(convertLectureDTOToLecture(lectureDTO));
+        return  convertLectureTOTDO(lecture1);
+//        course.getLectures().add(convertLectureDTOToLecture(lecture));
```
Comment (adamkulawik) [inline src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java:60]: remove this commented out code pls

File: src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java (hunk 1 above)
Comment (adamkulawik) [reply src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java:23]: can be private

File: src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java [hunk 6]
```diff
@@ -17,31 +18,48 @@
...
+
+    LectureDTO convertLectureTOTDO(Lecture lecture){
+        return  new LectureDTO(
+                lecture.getTitle(),
+                lecture.getDescription(),
+                lecture.getPrice(),
+                lecture.getDuration()
+        );
+    }
+    Lecture convertLectureDTOToLecture(LectureDTO lectureDTO){
```
Comment (adamkulawik) [inline src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java:31]: can be private

File: src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java [hunk 7]
```diff
@@ -40,38 +40,38 @@ Lecture convertLectureDTOToLecture(LectureDTO lectureDTO){
...
+        if ( lectureDTO.getTitle().length() >= titleLengthLimit ) {
             throw new IllegalLengthException();
         }
-        if ( lecture.getDescription().length() >= descriptionLengthLimit ) {
+        if ( lectureDTO.getDescription().length() >= descriptionLengthLimit ) {
             throw new IllegalLengthException();
         }
-        Lecture lectures = lectureRepository.save(convertLectureDTOToLecture(lecture));
-        return  convertLectureDTOToLecture(lecture);
+        Lecture lecture1 = lectureRepository.save(convertLectureDTOToLecture(lectureDTO));
```
Comment (LukaszGryziewicz) [inline src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java:58]: check formatting please

File: src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java (hunk 6 above)
Comment (Graczowsky) [inline src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java:31]: Mr. Adam please make those methods private, thanks

File: src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java [hunk 8]
```diff
@@ -40,38 +40,38 @@ Lecture convertLectureDTOToLecture(LectureDTO lectureDTO){
...
+        return  convertLectureTOTDO(lecture1);
```
Comment (LukaszGryziewicz) [inline src/main/java/com/coursesPlatform/coursePortfolio/LectureService.java:59]: check this method you're calling here to convertLectureToDTO or convertToDTO

**Pull request merged**